app/api/tasks.py

```diff
@@ -42,10 +49,45 @@ def get_and_check_project(self, request, project_pk, perms = ('view_project', ))
         except ObjectDoesNotExist:
             raise exceptions.NotFound()
         return project
-    
+
+    @detail_route(methods=['post'])
+    def cancel(self, request, pk=None, project_pk=None):
+        self.get_and_check_project(request, project_pk, ('change_project',))
+        try:
+            task = self.queryset.get(pk=pk, project=project_pk)
+        except ObjectDoesNotExist:
+            raise exceptions.NotFound()
+
+        task.pending_action = task.PendingActions.CANCEL
+        task.last_error = None
+        task.save()
+
+        # Call the scheduler (speed things up)
+        scheduler.process_pending_tasks(background=True)
```

Comment:
In essence it is replicating some of the Celery features (it is a task processing queue). What is driving me to re-implement this mechanism is that so far, the task processing involves more than a single processing step (think input images in ---> assets out). Maybe I'm over-thinking it, but so far I found these cases where Celery does not seem like it would fare so well.

1. Input images in --> assets are not what I expected. I should be able to restart the process without re-sending the images, just by changing the processing options?

2. Console output: how to retrieve real-time console output from the opendronemap process?

3. How to protect processing nodes (authentication) without making it difficult to setup? Ideally a processing node should be (optionally) protected by a static token, or a userID/password combination that generates a temporary token. The docs here hint at using public/private key exchange for security: http://docs.celeryproject.org/en/latest/userguide/security.html (I'm not sure I'd want to go down that route)